02-esc/challenges.js

```diff
@@ -1,143 +1,294 @@
-                setRating_star_3.classList.add('active');
-                setRating_star_4.classList.add('active');
-                setRating_star_5.classList.add('active'); 
-            }
-            
-            challenge_item.appendChild(setRating);
-            
-            let setTitle = document.createElement("h3");
-            setTitle.classList.add('challenge-title');
-            setTitle.textContent = title;
-            challenge_item.appendChild(setTitle);
-
-            let setParticipants = document.createElement("small");
-            setParticipants.classList.add('challenge-meta');
-            setParticipants.innerHTML = `${minParticipants} - ${maxParticipants} participants`;
-            challenge_item.appendChild(setParticipants);
-
-            let setDescription = document.createElement("p");
-            setDescription.classList.add('challenge-description');
-            setDescription.textContent = description;
-            challenge_item.appendChild(setDescription);
-
-            let btnBook = document.createElement("button");
-            btnBook.classList.add('btnBook');
-
-            btnBook.setAttribute("data-id", id); 
-                
-                if (type == 'onsite') {
-                    btnBook.textContent = 'Take challenge online';
-                } else {
-                    btnBook.textContent = 'Book this room';
-                }
-            challenge_item.appendChild(btnBook);
-
-            btnBook.addEventListener("click", function(){
-                console.log(id);
-                // referens till bokningsfunktion
-            });
-
-    
-            challenge_list.appendChild(challenge_item);
-            
-            console.log(title);
-      
-    };
-
-
-
-}  
\ No newline at end of file
+
+const challenge_list = document.querySelector(".challenge-list");
+const openFilterBtn = document.querySelector("#btnFilterChallenges");
+const closeFilterBtn = document.querySelector(".filter-close");
+
+openFilterBtn.addEventListener("click", () => {
+	filterForm.classList.toggle("open");
+	if (filterForm.classList.contains("open")) {
+		openFilterBtn.style.display = "none";
+	}
+});
+
+closeFilterBtn.addEventListener("click", () => {
+	filterForm.classList.toggle("open");
+	if (!filterForm.classList.contains("open")) {
+		openFilterBtn.style.display = "block";
+	}
+});
+
+const filterForm = document.querySelector(".filter-form");
+const searchFilterInput = document.querySelector("input[type='search']");
+const clearFilterBtn = document.querySelector(".filter-clear");
+
+clearFilterBtn.addEventListener("click", clearFilter);
+filterForm.addEventListener("submit", doNotReload);
+filterForm.addEventListener("change", useAllFilters);
+searchFilterInput.addEventListener("keyup", useAllFilters);
+
+let resultFromAPI = [];
+
+window.addEventListener("load", getChallengesAPI);
+
+async function getChallengesAPI() {
+	let tagsCollection = new Set();
+	resultFromAPI.length = 0;
```

Comment:
Nu känner jag mig trög här, men varför behöver denna sättas till 0? Är det för att den behöver tömmas inför filtrering?

Reply:
Nej, det är nog jag som råkat glömma kvar den raden när jag försökte få två filer att fungera ihop, innan jag bara skrev in allt mitt i din js-fil. Så den här raden behövs inte här, du har helt rätt 😁